### ISSUE TITLE
Issue with ember install on ember-cli v2.13.0-beta.3 on Windows

### ISSUE THREAD

**Issue by pixelhandler** [emberwatch/ember-cli-typescript#17]
The output from running `ember install ember-cli-typescript` …

```
C:\src\try-typescript [master +1 ~1 -0 !]> ember install ember-cli-typescript
Yarn: Installed ember-cli-typescript
installing ember-cli-typescript
Path must be a string. Received undefined
TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7:11)
    at Object.join (path.js:468:7)
    at Class.buildFileInfo (C:\src\try-typescript\node_modules\ember-cli\lib\models\blueprint.js:651:24)
    at Array.map (native)
    at Class._getFileInfos (C:\src\try-typescript\node_modules\ember-cli\lib\models\blueprint.js:678:18)
    at Class.processFiles (C:\src\try-typescript\node_modules\ember-cli\lib\models\blueprint.js:725:26)
    at tryCatch (C:\src\try-typescript\node_modules\rsvp\dist\rsvp.js:539:12)
    at invokeCallback (C:\src\try-typescript\node_modules\rsvp\dist\rsvp.js:554:13)
    at publish (C:\src\try-typescript\node_modules\rsvp\dist\rsvp.js:522:7)
    at flush (C:\src\try-typescript\node_modules\rsvp\dist\rsvp.js:2414:5)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
```

`ember install …` did use `yarn` to install, I also confirmed that I can `npm install ember-cli-typescript` as well. However the blueprint doesn't run.

```
C:\src\try-typescript [master +1 ~1 -0 !]> ember g ember-cli-typescript
installing ember-cli-typescript
Path must be a string. Received undefined
TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7:11)
    at Object.join (path.js:468:7)
    at Class.buildFileInfo (C:\src\try-typescript\node_modules\ember-cli\lib\models\blueprint.js:651:24)
    at Array.map (native)
    at Class._getFileInfos (C:\src\try-typescript\node_modules\ember-cli\lib\models\blueprint.js:678:18)
    at Class.processFiles (C:\src\try-typescript\node_modules\ember-cli\lib\models\blueprint.js:725:26)
    at tryCatch (C:\src\try-typescript\node_modules\rsvp\dist\rsvp.js:539:12)
    at invokeCallback (C:\src\try-typescript\node_modules\rsvp\dist\rsvp.js:554:13)
    at publish (C:\src\try-typescript\node_modules\rsvp\dist\rsvp.js:522:7)
    at flush (C:\src\try-typescript\node_modules\rsvp\dist\rsvp.js:2414:5)
```

Version info:
```
C:\src\try-typescript [master +1 ~1 -0 !]> ember --version
ember-cli: 2.13.0-beta.3
node: 6.10.0
os: win32 x64
```

Also I noticed that I was able to setup a repo using ember-cli-typescript with an ember app using ember-cli: 2.13.0-beta.3 on a Mac. Then when cloning the repo on Windows I had trouble running the app. There were a bunch of permissions issues. I have a project on an older version of ember-cli so tried that too and had similar build/serve issues…

```
The Broccoli Plugin: [SourceMapConcat: Concat: Vendor /assets/vendor.js] failed with:
Error: EPERM: operation not permitted, symlink

The Broccoli Plugin: [SourceMapConcat: Concat: Vendor /assets/vendor.js] failed with:
Error: EPERM: operation not permitted, lstat

The Broccoli Plugin: [BroccoliMergeTrees] failed with:
Error: EPERM: operation not permitted, symlink

The Broccoli Plugin: [Funnel: Funnel (lint app)] failed with:
Error: EPERM: operation not permitted, symlink
```

**Comment by pixelhandler** [comment]
@DanielRosenwasser Have you every tried TypeScript with ember-cli for a .Net project? I'm hoping to use TypeScript with Ember and deliver to a production system using a Windows based build system, e.g. TeamCity, Octopus Deploy etc. I'm pretty sure the build with work with this library, but I do have problems running the test suite with Chrome or Edge.

**Comment by 1bytee** [comment]
Doesn't work with ember 2.11.0 either.
```
installing ember-cli-typescript
Path must be a string. Received undefined
TypeError: Path must be a string. Received undefined
```
Command: `ember install "ember-cli-typescript@emberwatch/ember-cli-typescript"
`

**Comment by DanielRosenwasser** [comment]
Hi guys, I've not worked with Ember in the context of a TypeScript project - and to be quite honest, I am no Ember expert. 😃

However I have worked with the Ember core team so that TypeScript can actually properly model the Ember API. What I'd hope is that Ember users can take advantage of this in the near future and get the experience that Tom, Yehuda, and others on the core team have enjoyed. In the mean time, there is [Glimmer](https://glimmerjs.com/) (the component view library used in Ember).

While I can't definitely say whether Glimmer'll serve **all** your needs, I tried it the day it came out and it supported TypeScript from the ground up as a first-class citizen. I would definitely give it a shot if you're open to trying it.

**Comment by DanielRosenwasser** [comment]
Disregard my last message - I misread and thought you were having trouble with the library itself, not the installer. @pixelhandler it sounds like you've gotten things working apart from the test suite. Could you elaborate how?

**Comment by billheaton-q2** [comment]
@DanielRosenwasser thanks for the reply :) The TypeScript library seems fine. In this repo we have tooling for bring TypeScript to Ember CLI apps.

So far I've only been able to work around the install by manually creating the `tsconfig.json` file since the generator that automates this is broken on Windows (that's a blueprint for creating the config file in this repo).

After I manually create the `tsconfig.json` file I can run the tests for a newly generated app. But cannot run the server that ships with Ember CLI for developing the Ember application. Even though the build and the tests work the app can't be served up and run locally, the `ember server` command errors...

```
Error: ENOTEMPTY: directory not empty, rmdir 'C:\src\try-typescript\tmp\funnel-output_path-xz25SPVR.tmp'
...
Error: EPERM: operation not permitted, lstat 'C:\src\try-typescript\tmp\config_replace-input_base_path-B5Ck7urT.tmp\0'
```

I tried to check if there was a "read only" issue with node_modules directory but changing that didn't help.

Generally we need a solution for compatibility for developing Ember apps using TypeScript on Windows. My work around was using a Mac, this repo does work on the Mac.

**Comment by pixelhandler** [comment]
@DanielRosenwasser thanks for the reply :) The TypeScript library seems fine. In this repo we have tooling to bring TypeScript to Ember CLI apps.

So far I've only been able to work around the install issue by manually creating the `tsconfig.json` file; since the generator that automates this is broken on Windows (that's a blueprint for creating the config file in this repo).

After I manually create the `tsconfig.json` file, I can run the default tests for a newly generated Ember app. But I cannot run the server that ships with Ember CLI; used for developing the Ember application. Even though the build and the tests work, the app can't be served up and run locally, the `ember server` command errors...

```
Error: ENOTEMPTY: directory not empty, rmdir 'C:\src\try-typescript\tmp\funnel-output_path-xz25SPVR.tmp'
...
Error: EPERM: operation not permitted, lstat 'C:\src\try-typescript\tmp\config_replace-input_base_path-B5Ck7urT.tmp\0'
```

I tried to check if there was a "read only" issue with node_modules directory but changing that didn't help.

Generally we need a solution for compatibility for developing Ember apps using TypeScript on Windows. My work around was using a Mac, this repo does work on the Mac.
